connect/api/src/main/java/org/apache/kafka/connect/source/SourceRecord.java

```diff
@@ -46,25 +46,49 @@
     private final Map<String, ?> sourcePartition;
     private final Map<String, ?> sourceOffset;
 
+    /**
+     * @deprecated use {@link SourceRecord.Builder}
+     */
+    @Deprecated
```

Comment:
Good points. Especially agree about the 2nd and 3rd con. I'll remove the deprecations and the Builder.

Do you think we should have timestamp variants for all the current constructors, or only a new constructor which takes all of the args? It feels like specialized usage to me so I think the latter makes more sense.